diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter3/CollapseIfStatementsRule.kt

```diff
@@ -85,33 +84,46 @@ class CollapseIfStatementsRule(configRules: List<RulesConfig>) : DiktatRule(
     private fun findNestedIf(parentNode: ASTNode): ASTNode? {
         val parentThenNode = (parentNode.psi as KtIfExpression).then?.node ?: return null
         val nestedIfNode = parentThenNode.findChildByType(IF) ?: return null
-        // Nested `if` node should be the last child, but actually,
-        // the last children are WHITESPACE and `}`, so take treePrev
-        if ((nestedIfNode.psi as KtIfExpression).node != parentThenNode.lastChildNode.treePrev.treePrev) {
-            return null
-        }
-        // We won't collapse, if statements some of them have `else` node
+        // We won't collapse if-statements, if some of them have `else` node
         if ((parentNode.psi as KtIfExpression).`else` != null ||
                 (nestedIfNode.psi as KtIfExpression).`else` != null) {
             return null
         }
-        // We monitor which types of nodes are followed before nested `if`
+        // We monitor which types of nodes are followed before and after nested `if`
         // and we allow only a limited number of types to pass through.
         // Otherwise discovered `if` is not nested
         val listOfNodesBeforeNestedIf = parentThenNode.getChildren(null).takeWhile { it.elementType != IF }
-        val allowedTypes = listOf(LBRACE, WHITE_SPACE, RBRACE, KDOC, BLOCK_COMMENT, EOL_COMMENT)
-        if (listOfNodesBeforeNestedIf.any { it.elementType !in allowedTypes }) {
+        val listOfNodesAfterNestedIf = parentThenNode.getChildren(null).takeLastWhile { it != parentThenNode.findChildByType(IF) }
```

Comment:
Why isn't `elementType` check enough here? Also, how will it work in case of several independent `if`s in the same block?

Reply:
Previously, there were the check for nested if: we assumed, that current `if-statement` is nested, if **it is** the last child for external `if (not considered whitespaces and brackets)

Now, by analogy `listOfNodesBeforeNestedIf,` we suppose, that nested if not only the last child, but more - after nested `if` it also could be comments
In this case we can collapse statements too

Reply:
If there will be several independent if-statements, like this
```kotlin
if (cond1) {
    if (cond2) {
    }
    if (cond3) {
    }
}
```
we won't collapse anything, since `if (cond2)` for the first look seems to be nested, but after this node we can see another IF,
which is not any type of `LBRACE, WHITE_SPACE, RBRACE, BLOCK_COMMENT, EOL_COMMENT`
And it's correct behavior, since we shouldn't collapse it like this:
```kotlin
if (cond1 && cond2) {
    if (cond3) {
    }
}
```

Reply:
Yes, I remember that we discussed it, but I've mixed the details. Alright then